vm.go

```diff
@@ -171,10 +171,10 @@ func intToValue(i int64) Value {
 	}
 	return valueFloat(i)
 }
-func int64toValue(i int64) Value {
+func int64ToValue(i int64) Value {
 	if i >= -maxInt && i <= maxInt {
 		if i >= -128 && i <= 127 {
-			return intCache[i+128]
+			return int64Cache[i+128]
```

Comment:
this was the core of the change, we previously used the intCache which would return a `valueInt` not a `valueInt64`